README_ Creation (#2405)
Techtonica/curriculum: projects/recipe-page/README.md

# Recipe Page

## 📌 Prerequisites
- Basic HTML and CSS
- Bootstrap or any CSS framework
- DOM manipulation (i.e. jQuery)
- Git/GitHub basics

> ## 🎯 Goal of the Project

Reviewer: ```suggestion
## 🎯 Purpose of the Project
```

 A fully responsive recipe webpage 

## ✅ Project Requirements
- List of key deliverables for the project like Recipe Content:Title and image of the dish ,List of ingredients, Step-by-step instructions,Cooking time and servings

- HTML structure
- CSS styling
- Responsive design using Bootstrap
- Functional navigation qith javascript

## 🗓️ FT Program Weekly Breakdown
| Phase | Description |
|-------|-------------|

> | Phase 1 | Create structure with HTML |
> | Phase 2 |  Add styling with CSS |

Reviewer: ```suggestion
| Phase 1 | Create structure with HTML |
| Phase 2 | Add styling with CSS |
```

| Phase 3 |Incorporate the Bootstrap Framework |
| Phase 4 |Incorporate JS DOM |

> ## 📁 Folder/File Reference
> | File/Folder | Purpose |
> |-------------|---------|
> | index.html | Main HTML file |
> | styles.css | Custom CSS styling |
> | /screenshots | Folder for images |

Reviewer: ```suggestion
```

remove

> ## 🏁 Project Milestones
> - [ ] Setup folder and tools
> - [ ] Complete HTML layout
> - [ ] Apply CSS styling
> - [ ] Add responsive features (Bootstrap)
> - [ ] DOM manipulation (i.e. jQuery)
> - [ ] Submit and review

Reviewer: ```suggestion
## 🏁 Project Goals
- Add responsive features using CSS (or a CSS framework such as Bootstrap)
- Learn how to access and manipulate elements using JS
- Practice incorporating DOM manipulation (such as jQuery)
- Add content via HTML layout 
- Add a check mark as an element to a list
- Add a strikethrough when someone clicks an element inside a checkbox
- Submit a PR and participate in a code review
```

Author: done

## Week 1 Milestones Overview

### Phase 1 - Create structure with HTML

### Basic Requirements 

#### 1. HTML Structure:

- Create an HTML file with a proper structure

- Use `<div>` tags to wrap sections for layout control

#### 2. Recipe Content:

- Add a title and a short background of the recipe

- Include an unordered list of ingredients

- Include an ordered list for preparation steps

- Add at least one image of the dish

- Link to related/similar recipes online

#### 3. Version Control:

- Make at least 15 commits throughout development

- Submit a pull request and ensure it's reviewed by a mentor

 
### Phase 2 - Add styling with CSS

**Basic Requirements**

#### 1. Project Structure:

- Set up a proper file tree for organization

- Create a CSS file and link it in the HTML `<head>`

#### 2. Styling:

- Change the background color of the page

- Customize fonts and font colors for headings and paragraphs

> #### 3. Layout:
> 
> - Constrain the description width to match the image
> 
> - Display the image and ingredients on the left
> 
> - Display the instructions on the right
> 
> 
> ## Week 2  Milestones Overview
> 
> ### Phase 3 - Incorporate the Bootstrap Framework
> 
> ### Basic Requirements
> 
> 1. Properly link to the Bootstrap CSS CDN
> 2. Add column classes to your divs to make the webpage responsive
> 3. Add containers for the navigation bar and content
> 4. Add a navigation bar to the top of your webpage
> 
> ## Setup Instructions
> 
> - Navigate to the `recipe-page` directory.
> 
> ```shell
> cd recipe-page
> ```
> 
> ### Phase 4 - Incorporate JS DOM
> 
> #### Languages
> 
> - HTML
> - CSS
> - Javascript
> 
> #### Primary Goals
> 
> - [ ] Learn how to access and manipulate elements using JS
> - [ ] Practice incorporating JS DOM methods
> 
> ### Overview
> 
> #### 1. Adding Content
> 
> #### 2. Adding a check mark as an element to a list
> 
> #### 3. Adding a strikethrough when someone clicks an element
> 
> #### 4. Adding a strikethrough when someone clicks an element inside a checkbox
> 
> 
> 
> ## ❓ General FAQ
> **Q:** What should I do if I’m stuck during a phase?  
> **A:** Spend at least 20 minutes troubleshooting, then ask for help with a detailed question.
> 
> **Q:** What is the final deliverable?  
> **A:** A complete responsive web page or portfolio with interactivity.
> 

Reviewer: ```suggestion
## Week 1 Milestones

**Phase 1: Create structure with HTML**
1. HTML Structure:
  - Create an HTML file with a proper structure
  - Use `<div>` tags to wrap sections for layout control
2. Recipe Content:
  - Add a title and a short background of the recipe
  - Include an unordered list of ingredients
  - Include an ordered list for preparation steps
  - Add at least one image of the dish
  - Link to related/similar recipes online
3. Version Control:
  - Make at least 15 commits throughout development
  - Submit a pull request and ensure it's reviewed by a mentor

**Phase 2: Add styling with CSS**
1. Project Structure:
  - Set up a proper file tree for organization
  - Create a CSS file and link it in the HTML `<head>`
2. Styling:
  - Change the background color of the page
  - Customize fonts and font colors for headings and paragraphs
3. Layout:
  - Constrain the description width to match the image
  - Display the image and ingredients on the left
  - Display the instructions on the right

## Week 2  Milestones Overview

**Phase 3: Incorporate the Bootstrap Framework**
1. Properly link to the Bootstrap CSS CDN
2. Add column classes to your divs to make the webpage responsive
3. Add containers for the navigation bar and content
4. Add a navigation bar to the top of your webpage
```

# Recipe Page

## 📌 Prerequisites

> - Basic HTML, CSS (for Recipe)
> - Bootstrap, jQuery (for advanced phases)
> - Git/GitHub basics
> - [Any tools like VS Code, Terminal, etc.]

Reviewer: ```suggestion
- Basic HTML and CSS
- Bootstrap or any CSS framework
- DOM manipulation (i.e. jQuery)
- Git/GitHub basics
```

## 🎯 Goal of the Project
 A fully responsive recipe webpage 

## ✅ Project Requirements

> - [List of key deliverables for the project]

Reviewer: you have not summarized the key requirements, but rather have a placeholder here

- HTML structure
- CSS styling
- Responsive design using Bootstrap
- Functional navigation qith javascript

> - [Interactive element if required]

Reviewer: you have not summarized the key interactive elements, but rather have a placeholder here

> ## 🗓️ FT Program Weekly Breakdown
> | Week | Phase | Description |
> |------|-------|-------------|
> | Week 1 | Phase 1 | Create structure with HTML |
> | Week 1 | Phase 2 |  Add styling with CSS |
> | Week 2 | Phase 3 |Incorporate the Bootstrap Framework |
> | Week 2 | Phase 4 |Incorporate JS DOM |
> 
> ## 📁 Folder/File Reference
> | File/Folder | Purpose |
> |-------------|---------|
> | index.html | Main HTML file |
> | styles.css | Custom CSS styling |
> | /img | Folder for images |
> | /scripts.js | Any JS or jQuery scripts |

Reviewer: ```suggestion
## 🗓️ Project Breakdown
| Phase | Description |
|-------|-------------|
| Phase 1 | Create structure with HTML |
| Phase 2 |  Add styling with CSS |
| Phase 3 | Incorporate Bootstrap or Another Framework |
| Phase 4 | Incorporate JS DOM |

```

## 🏁 Project Milestones
- [ ] Setup folder and tools
- [ ] Complete HTML layout
- [ ] Apply CSS styling
- [ ] Add responsive features (Bootstrap)

> - [ ] Add interactivity (jQuery)

Reviewer: ```suggestion
- [ ] DOM manipulation (i.e. jQuery)
```

Author: done

- [ ] Submit and review

> ## 🧰 Prep Work
> - Review: HTML/CSS basics
> - Explore Bootstrap components: Navbar, Grid, Cards
> - Read jQuery intro tutorial (if applicable)

Reviewer: ```suggestion
```

> # Week 1 Milestones Overview

Reviewer: Lines 49 onward, in this document are simply a copy and paste of each phase, this is not the purpose of this README. It should be an introduction to the entire project concisely. Please remove the content lines 49+ and kindly identify the core requirements and summarize them, or link to each phase's respective requirement section in the "project requirements" section: 
- [Phase 1 requirements](https://github.com/Techtonica/curriculum/blob/main/projects/recipe-page/phase-1-html-prompt.md#basic-requirements)
- [Phase 2 requirements](https://github.com/Techtonica/curriculum/blob/main/projects/recipe-page/phase-2-css-prompt.md#basic-requirements)
...etc...

  - Add a title and a short background of the recipe
  - Include an unordered list of ingredients
  - Include an ordered list for preparation steps
  - Add at least one image of the dish
  - Link to related/similar recipes online
3. Version Control:
  - Make at least 15 commits throughout development
  - Submit a pull request and ensure it's reviewed by a mentor
**Phase 2: Add styling with CSS**
1. Project Structure:
  - Set up a proper file tree for organization
  - Create a CSS file and link it in the HTML `<head>`
2. Styling:
  - Change the background color of the page
  - Customize fonts and font colors for headings and paragraphs
3. Layout:
  - Constrain the description width to match the image
  - Display the image and ingredients on the left
  - Display the instructions on the right
## Week 2  Milestones Overview
**Phase 3: Incorporate the Bootstrap Framework**
1. Properly link to the Bootstrap CSS CDN
2. Add column classes to your divs to make the webpage responsive
3. Add containers for the navigation bar and content
4. Add a navigation bar to the top of your webpage

## Setup Instructions
- Navigate to the `recipe-page` directory.

```shell
cd recipe-page

> ```

Author: thanks for the suggestion
